integration-cli/docker_utils.go

```diff
@@ -1567,3 +1567,23 @@ func sumAsIntegers(vals ...interface{}) interface{} {
 	}
 	return s
 }
+
+// pollFunc is used to periodically execute a check function, it
+// returns error after timeout.
+func pollFunc(f func() error, timeout time.Duration) error {
```

Comment:
I tried and sadly wasn't able to write proper code :( I think waitAndAssert too magical and requires non-obvious code to write.